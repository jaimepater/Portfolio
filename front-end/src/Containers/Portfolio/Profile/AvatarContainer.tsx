import * as React from 'react';
import styled from 'styled-components';
import Avatar from '../../../Components/Avatar';

const Container = styled.div`
  display: flex;
  justify-content: center;
`;

export const AvatarBorder = styled.div`
  width: 180px;
  height: 180px;
  display: flex;
  border-radius: 50%;
  padding: 7px;
  background-color:  ${props => props.theme.palette.white};
  border: 3px solid   ${props => props.theme.palette.blue1};
  z-index: 2;
`;

const AvatarContainer = () => (
  <Container>
    <AvatarBorder>
      <Avatar />
    </AvatarBorder>
  </Container>
);

export default AvatarContainer;
