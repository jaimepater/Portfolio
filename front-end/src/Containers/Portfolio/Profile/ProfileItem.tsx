import { default as React } from 'react';
import styled, { DefaultTheme, keyframes } from 'styled-components';
import H2 from '../../../Components/H2';

const fadeUp = keyframes`
  from { opacity: 0; }
  to { opacity: 1; }
`;


// Create the keyframes
const animation = keyframes`
 from {
    opacity: 0;
    -webkit-transform: translate3d(-100%, 0, 0);
    transform: translate3d(-100%, 0, 0);
  }
  20% {
   opacity: 1;
    -webkit-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }
  70% {
   opacity: 0.5;
  }
  to{
  opacity: 0;
  }
`;
export const TextProfile = styled(H2)`
 font-family: 'Roboto';
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
      line-height: normal;
  animation: ${animation} ${(props: ITextProfile) => props.time} linear;
  animation-iteration-count: initial;
  letter-spacing: normal;
  text-align: center;
  margin-top: 0;
  color: ${props => props.theme.palette.gray1};
`;

interface IProfileItemProps {
  profile: string,
  time: number,
}

interface ITextProfile {
  theme: DefaultTheme,
  time: string,
}

const ProfileItem = ({ profile, time } : IProfileItemProps) => {
  const animationTime = `${time / 1000}s`;
  return (<TextProfile time={animationTime}>{profile}</TextProfile>);
};


export default ProfileItem;
