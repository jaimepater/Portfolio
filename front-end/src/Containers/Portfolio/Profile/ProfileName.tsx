import * as React from 'react';
import styled, { keyframes } from 'styled-components';
import { Grid } from '@material-ui/core';
import H1 from '../../../Components/H1';
import Colombia from '../../../Components/Colombia';



export const Name = styled.div`
 font-family: 'Roboto';
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: normal;
  letter-spacing: normal;
  text-align: left;
  color: ${props => props.theme.palette.gray1};
`;

const Flag = styled.div`
    height: 30px;
    width: 40px;
`;


interface IProfileNameProps {
  name: string
}

const ProfileName = ({ name } : IProfileNameProps) => (
  <Grid
    container
    direction="row"
    justify="center"
    alignItems="center"
    spacing={1}
  >
    <Grid item>
      <Flag>
        <Colombia />
      </Flag>
    </Grid>
    <Grid item>
      <Name>
        <H1>
          {name}
        </H1>
      </Name>
    </Grid>
  </Grid>
);

export default ProfileName;
