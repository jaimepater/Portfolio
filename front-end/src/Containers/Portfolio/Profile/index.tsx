import * as React from 'react';
import { observer } from 'mobx-react';
import { useEffect } from 'react';
import useStores from '../../../Commons/Hooks/Stores';
import Profile from '../../../Components/Profile/Profile';

const ProfilePortfolio = observer(() => {
  const { profileStore } = useStores();
  useEffect(() => {
    profileStore.getProfileData();
  }, [profileStore]);

  const { fullName, profiles } = profileStore;
  const profile = {
    name: fullName,
    showFlag: true,
    list: profiles,
  };
  return <Profile {...profile} />;
});

export default ProfilePortfolio;
