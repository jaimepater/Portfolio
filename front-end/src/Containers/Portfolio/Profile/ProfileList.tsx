import * as React from 'react';
import { useEffect, useState } from 'react';
import ProfileItem from './ProfileItem';

interface IProfileListProps {
  profiles: Array<string>;
  time: number;
}

const ProfileList = ({ profiles, time }: IProfileListProps) => {
  const [profile, setProfile] = useState({ id: 0, name: '' });
  const showProfile = (index: any) => {
    if (!isNaN(index)) {
      setProfile({ name: profiles[index], id: index });
      setTimeout(showProfile.bind(undefined, (index + 1) % profiles.length), time);
    }
  };

  useEffect(() => {
    showProfile(0);
  }, [profiles, profiles.length, time]);

  return <ProfileItem key={profile.id} profile={profile.name} time={time} />;
};

export default ProfileList;
