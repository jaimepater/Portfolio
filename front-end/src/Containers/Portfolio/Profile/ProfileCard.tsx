// @flow
import * as React from 'react';
import {
  Card, CardActionArea, CardContent, Typography,
} from '@material-ui/core';
import styled from 'styled-components';
import { useEffect, useState } from 'react';
import { inject, observer } from 'mobx-react';
import { IProfileStore } from '../../../Commons/Stores/ProfileStore';
import ProfileName from './ProfileName';


const StyledCard = styled(Card)`
  margin-top: -50px;
`;

const StyledCardContent = styled(CardContent)`
  padding-top: 55px;
`;

interface CardProps {
  profileStore? : IProfileStore
}
const ProfileCard: React.FC<CardProps> = ({ profileStore } : CardProps) => {
  useEffect(() => {
    console.log('hiiiiiii');
    profileStore && profileStore.getProfileData();
    console.log(profileStore && profileStore.profileData.name);
  }, [profileStore]);


  return (
    <StyledCard>
      <StyledCardContent>
        <div>
          <ProfileName name={profileStore && profileStore.profileData.name} />
        </div>
      </StyledCardContent>
    </StyledCard>
  );
};

export default inject('profileStore')(observer(ProfileCard));
