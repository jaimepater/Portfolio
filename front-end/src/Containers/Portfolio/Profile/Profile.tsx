import * as React from 'react';
import { Container } from '@material-ui/core';
import styled from 'styled-components';
import AvatarContainer from './AvatarContainer';
import ProfileCard from './ProfileCard';

const ProfileContainer = styled.div`
  padding-top: 5%;
`;

const Profile = () => (
  <Container fixed>
    <ProfileContainer>
      <AvatarContainer />
      <ProfileCard />
    </ProfileContainer>
  </Container>
);

export default Profile;
