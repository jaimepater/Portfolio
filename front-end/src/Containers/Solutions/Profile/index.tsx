import * as React from 'react';
import { observer } from 'mobx-react';
import { useEffect } from 'react';
import styled from 'styled-components';
import useStores from '../../../Commons/Hooks/Stores';
import Profile from '../../../Components/Profile/Profile';
import { StyledCard } from '../../../Components/Profile/ProfileCard';
import H1 from '../../../Components/H1';
import H2 from '../../../Components/H2';
import Logo from '../../../Components/Logo';

const StyledProfile = styled.div`
  padding-top: 64px;
  & .avatarContainer {
    display: none;
  }
  ${StyledCard} {
    margin-top: 0;
  }
  ${H1}, ${H2} {
    color: ${props => props.theme.palette.white};
  }
  & .MuiCard-root {
    background-color: transparent;
    box-shadow: none;
  }
`;

const StyledLogoContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #3f51b5;
  border-radius: 50%;
  height: 40px;
  width: 40px;
  margin-right: 10px;
`;

const ProfileSolutions = observer(() => {
  const { solutionStore } = useStores();
  useEffect(() => {
    solutionStore.getProfileData();
  }, [solutionStore]);

  const { name, profiles } = solutionStore;
  const profile = {
    name,
    flag: (
      <StyledLogoContainer>
        <Logo />
      </StyledLogoContainer>
    ),
    list: profiles,
  };
  return (
    <StyledProfile>
      <Profile {...profile} />
    </StyledProfile>
  );
});

export default ProfileSolutions;
