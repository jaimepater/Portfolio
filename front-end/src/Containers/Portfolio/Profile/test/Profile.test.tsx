import React from "react";
import {shallow} from 'enzyme';
import Profile from "../Profile";



describe('Profile', () => {
  it('Renders Profile', () => {
    const shallowComponent = shallow(<Profile />);
    expect(shallowComponent).toMatchSnapshot();
  });
});