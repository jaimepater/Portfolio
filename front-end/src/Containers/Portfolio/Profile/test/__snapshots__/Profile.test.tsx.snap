// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Renders Profile 1`] = `
<WithStyles(ForwardRef(Container))
  fixed={true}
>
  <styled.div>
    <AvatarContainer />
    <Memo(wrappedComponent) />
  </styled.div>
</WithStyles(ForwardRef(Container))>
`;
